internal/controller/acl/acl.go

```diff
@@ -81,21 +80,22 @@ func Setup(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter) error {
 		Complete(r)
 }
 
-// A connector is expected to produce an ExternalClient when its Connect method
-// is called.
-type connector struct {
+// A connectDisconnector is expected to produce an ExternalClient when its Connect method
+// is called and close it when its Disconnect method is called.
+type connectDisconnector struct {
 	kube         client.Client
 	usage        resource.Tracker
 	log          logging.Logger
 	newServiceFn func(ctx context.Context, creds []byte, kube client.Client) (*kadm.Client, error)
+	cachedClient *kadm.Client
```

Comment:
This approach isn't thread-safe. I don't believe the reconciler loop runs multi-threaded, but if it does this approach may result in trouble. Simultaneous reconciliations could result in a client being prematurely closed, or unexpected replaced by a client pointing to the wrong Kafka cluster.

I would definitely like some feedback here. I can look into wrapping this with a mutex to make simultaneous access impossible.

Reply:
I don't believe the reconciler is multi-threaded, but it has been a while since I have looked.  I asked @hasheddan and/or @turkenh to take a look just to be sure.

Reply:
No, it looks like it is not multi-threaded.  https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller@v0.9.6#Options

We're just using the default here, which is 1.  To your point though, if we wanted to have concurrency here, it would be a problem.  However, I want to get this PR merged and isolate that as an enhancement.  The changes you made here are better to be added asap, and then we can look at concurrency in the future as needed.